### ISSUE TITLE
Add ability to customize the hidden dir path

### ISSUE THREAD

**Issue by next-jesusmanuelnavarro** [jmnavarrol/bash-magic-enviro#8]
BME tracks status, some project configurations, etc. under the hardcoded directory "${BME_PROJECT_DIR}/${BME_HIDDEN_DIR}/" (and also under "~/${BME_HIDDEN_DIR}/").

This has shown to be a problem under one usecase of mine: when taking advantage of the *python virtualenvs* module **and** sharing a project root's directory between a host and a container, it may happen that the built virtualenv is not compatible between both so when entering into the project space from the host, it creates a virtualenv that can't be used within the container and vice versa.  I suppose this can also happen with other modules architecture-dependent, like the terraform one.

I think it'd be good if the project's hardcoded directory could be overwritten, say, by an environment variable, so the one from the host doesn't overlap with the one used within the container.  This could mean some more wasted space and maybe some other directory to be added to .gitignore but I consider this to be a reasonable trade off.

**WORKAROUND:** right now I delete the offending directory each time I start my container but this leads to virtualenvs having to be rebuilt each time I go in and out of my container.

**Comment by jmnavarrol** [comment]
A workaround:

Setting the environment variable BME_HIDDEN_DIR to a new value different to its '.bme.d' default should do the expected.

I.e.: within the project's `.bme_project file` you could do something like this (possibly adding some logic about when  to set or not the new value):

```sh
(...)
old_hidden_dir="${BME_HIDDEN_DIR}"
BME_HIDDEN_DIR='.bme.different'

bme_custom_clean() {
	bme_log "Custom cleaning..." info 1
	BME_HIDDEN_DIR="${old_hidden_dir}"
	unset old_hidden_dir
	bme_log "Custom cleaning finished" info 1
}
```

Since it's set at session start, when BME is first loaded, global BME_CONFIG_DIR value should be preserved (pointing to `~/.bme.d`) but the project's hidden dir would be set **before** any module is loaded so it would set and use the new value.

The value is properly restored before going out the project space with the help of `bme_custom_clean()` function.

**BEWARE:** While I couldn't find any drawback, *BME_HIDDEN_DIR* is **not** a *"public"* variable meant to be customized/overwritten, so there's no guarantee this won't break in future releases.  All I can offer is trying to preserve it and/or offer a suitable upgrade path if/when that happens.

**Comment by jmnavarrol** [comment]
After some more tests, at the very least the terraform module, maybe others too, doesn't behave properly: both tfenv and terraform modules install on the wrong paths:
* tfenv under `${BME_PROJECT_DIR}/.bme.d`
* terraform providers under `${BME_PROJECT_DIR}/.terraform`.

**Comment by jmnavarrol** [comment]
New code has been merged to *main*: now you can set the **BME_PROJECT_CONFIG_DIR** in your project file to overwrite its default under project's root .bme.d/ directory.

You can set this variable to either an absolute path or a relative to the project's root one.  The variable is reset once you go off your project's hierarchy.

I'll let it this change *"in the wild"* for a while before cutting a new version including it.

**Comment by jmnavarrol** [comment]
Closed [version v1.8.2](../releases/tag/v1.8.2) that offers the new **BME_PROJECT_CONFIG_DIR** variable setable in .bme_project file.
